tests/cases/conformance/types/unknown/unknownControlFlow.ts

```diff
@@ -402,3 +402,32 @@ function doSomething2(value: unknown): void {
         value;
     }
 }
+
+// Repro from #51009
+
+type TypeA = {
+    A: 'A',
+    B: 'B',
+}
+
+type TypeB = {
+    A: 'A',
+    B: 'B',
+    C: 'C',
+}
+
+type R<T extends keyof TypeA> =
+    T extends keyof TypeB ? [TypeA[T], TypeB[T]] : never
+
+type R2<T extends PropertyKey> =
+    T extends keyof TypeA ?
+    T extends keyof TypeB ?
+        [TypeA[T], TypeB[T]] : never: never

```

Comment:
nit:
```suggestion
        [TypeA[T], TypeB[T]] : never : never
```

note that accepting this would require regenerating the baselines